marshal/tests/serialization/set.go

```diff
@@ -0,0 +1,104 @@
+package serialization
+
+import (
+	"bytes"
+	"fmt"
+	"runtime/debug"
+	"testing"
+
+	"github.com/gocql/gocql/marshal/tests/funcs"
+	"github.com/gocql/gocql/marshal/tests/utils"
+)
+
+type Sets []*Set
+
+// Set is a tool for generating test cases of marshal and unmarshall funcs.
+// For cases when the function should no error,
+// marshaled data from Set.Values should be equal with Set.Data,
+// unmarshalled value from Set.Data should be equal with Set.Values.
+type Set struct {
+	Data   []byte
+	Values []interface{}
+
+	IssueMarshal   string
+	IssueUnmarshal string
+}
+
+func (s Set) Run(name string, t *testing.T, marshal func(interface{}) ([]byte, error), unmarshal func([]byte, interface{}) error) {
```

Comment:
Let's drop `name` from the `Run`
```suggestion
func (s Set) Run(t *testing.T, marshal func(interface{}) ([]byte, error), unmarshal func([]byte, interface{}) error) {
```

Reply:
If the name is deleted from `Set.Run`, then to understand which of the `Set.Run` has a problem, you will need to search in all similar `Set.Run`. The opportunity to briefly save the meaning of `Set.Run` is also lost. 
In cases of `serialization` for `nil`, `zero by len` , `zero by value` data will lead to confusion due to the abundance of different variants.
In cases of `MustFail`, there will be a complete misunderstanding, so in order to understand how the data is distorted, you first need to understand its non-distorted appearance, then compare it with the distorted one. 
For all `complex types`, this solution is a direct path to complete failure in evaluating the fullness of testing unmarshal on `MustFail`.
Do you really want these problems?

Reply:
Yes, let's do that, if you run tests against same data, wrap it with `t.Run`, it is going to be isolated to a separate run.
there is good solution to mustfail.